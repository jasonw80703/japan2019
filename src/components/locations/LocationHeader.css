.location-header-div {
  margin-bottom: 1.5%;
}

#home-link {
  color: white;
}

.current-location-item, .current-location-item:hover {
  background-color: lightgray;
}