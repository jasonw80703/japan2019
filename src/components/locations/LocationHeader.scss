.location-header-div {
  margin-bottom: 1.5%;
}

#navbar-header {
  background-color: #EAE7DC !important;
}

#navbar-brand {
  color: #E85A4F;
}

@media (max-width: 500px) {
  #navbar-dates {
    display: none;
  }
}

#home-link {
  color: #E98074;
}

.current-item, .current-item:hover {
  background-color: #D8C3A5;
}

#location-nav .dropdown-menu {
  background-color: #EAE7DC;
}

#location-nav .dropdown-menu a:hover {
  background-color: #D8C3A5;
}