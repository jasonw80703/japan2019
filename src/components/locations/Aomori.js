import React from 'react';
import { getS3ObjectImagePath } from 'helpers/getS3ObjectPath';
import LocationHeader from 'components/locations/LocationHeader';
import Page from 'components/blog/Page';
import Image from 'components/blog/Image';
import TextBlock from 'components/blog/TextBlock';
import TextHeader from 'components/blog/TextHeader';
import ImageSlide from 'components/blog/ImageSlide';
import { LOCATIONS } from 'constants/locations';

// Map
import map from 'assets/map/aomori.jpg';
import mapModalDetails from 'assets/map/aomori.json';

// Texts
import aomoriText from 'assets/texts/aomori.json';
import captions from 'assets/images/aomori/captions.json';

const LOCATION = LOCATIONS.AOMORI;

export default function Aomori() {
  return (
    <div>
      <LocationHeader
        location={LOCATION}
        map={map}
        mapModalDetails={mapModalDetails}
      />
      <Page location={LOCATION}>
        <TextBlock text={aomoriText.getting_there_1} source={getS3ObjectImagePath('aomori/starbucks')} location={LOCATION['en']} imageOnRight={false} />
        <TextBlock text={aomoriText.getting_there_2} source={getS3ObjectImagePath('aomori/train')} location={LOCATION['en']} />
        
        <TextHeader header={aomoriText.headers.explore} />
        <TextBlock text={aomoriText.explore_1} />
        <Image source={getS3ObjectImagePath('aomori/daikon')} size='sm' location={LOCATION['en']} caption={captions.daikon} />
        <TextBlock text={aomoriText.explore_2} />
        <Image source={getS3ObjectImagePath('aomori/water')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.explore_3} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/aspam_1'),
            getS3ObjectImagePath('aomori/aspam_2'),
            getS3ObjectImagePath('aomori/aspam_3'),
            getS3ObjectImagePath('aomori/aspam_4')
          ]}
          location={LOCATION['en']}
        />

        <TextHeader header={aomoriText.headers.museum} />
        <Image source={getS3ObjectImagePath('aomori/museum')} location={LOCATION['en']} caption={captions.museum} />
        <TextBlock text={aomoriText.museum_1} />
        <Image source={getS3ObjectImagePath('aomori/nebuta_1')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.museum_2} source={getS3ObjectImagePath('aomori/nebuta_2')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.museum_3} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/nebuta_3'),
            getS3ObjectImagePath('aomori/nebuta_4'),
            getS3ObjectImagePath('aomori/nebuta_5'),
            getS3ObjectImagePath('aomori/nebuta_6'),
            getS3ObjectImagePath('aomori/nebuta_7'),
            getS3ObjectImagePath('aomori/nebuta_8')
          ]}
          location={LOCATION['en']}
        />

        <TextBlock text={aomoriText.conc} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/end_1'),
            getS3ObjectImagePath('aomori/end_2'),
            getS3ObjectImagePath('aomori/end_3'),
            getS3ObjectImagePath('aomori/end_4'),
            getS3ObjectImagePath('aomori/end_5'),
            getS3ObjectImagePath('aomori/end_6')
          ]}
          location={LOCATION['en']}
        />

      </Page>
    </div>
  )
}
