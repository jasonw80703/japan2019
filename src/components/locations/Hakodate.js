import React from 'react';
import LocationHeader from './LocationHeader';
import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.HAKODATE;

export default function Hakodate() {
  return (
    <div>
      <LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} />
      <Page location={LOCATION['en']} locationJp={LOCATION['jp']}></Page>
    </div>
  )
}