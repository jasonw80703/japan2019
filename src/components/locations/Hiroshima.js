import React from 'react';
import LocationHeader from './LocationHeader';
// import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.HIROSHIMA;

export default function Hiroshima() {
  return (
    <div><LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} /></div>
  )
}