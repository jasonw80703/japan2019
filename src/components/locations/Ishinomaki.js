import React from 'react';
import LocationHeader from 'components/locations/LocationHeader';
import Page from 'components/blog/Page';
// import Image from 'components/blog/Image';
// import TextBlock from 'components/blog/TextBlock';
// import TextHeader from 'components/blog/TextHeader';
import { LOCATIONS } from 'constants/locations';

// Map
import map from 'assets/map/ishinomaki.jpg';
import mapModalDetails from 'assets/map/ishinomaki.json';

const LOCATION = LOCATIONS.ISHINOMAKI;

export default function Ishinomaki() {
  return (
    <div>
      <LocationHeader
        location={LOCATION}
        map={map}
        mapModalDetails={mapModalDetails}
      />
      <Page location={LOCATION}></Page>
    </div>
  )
}
