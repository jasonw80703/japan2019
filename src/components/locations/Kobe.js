import React from 'react';
import LocationHeader from './LocationHeader';
import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.KOBE;

export default function Kobe() {
  return (
    <div>
      <LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} />
      <Page location={LOCATION['en']} locationJp={LOCATION['jp']}></Page>
    </div>
  )
}