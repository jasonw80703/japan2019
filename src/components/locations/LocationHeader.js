import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';
import MapModal from './MapModal';
import { LOCATIONS } from '../../constants/locations';
import lowerCaseRemoveSpaces from '../../helpers/lowerCaseRemoveSpaces';
import './LocationHeader.css';

/**
 * Map locations constant to NavDropdown with an Item for each location
 */
function LocationNavDropdown() {
  return (
    <NavDropdown title="Location" id="location-dropdown">
      {
        LOCATIONS.map((location) => (
          <NavDropdown.Item href={lowerCaseRemoveSpaces(location)} key={location}>{location}</NavDropdown.Item>
        ))
      }
    </NavDropdown>
  );
};

// TODO:
// * add next button to go to next city
// * brand can be dropdown that shows current location
export default class LocationHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
    };

    this.handleToggleModal = this.handleToggleModal.bind(this);
    this.getLocationSource = this.getLocationSource.bind(this);
  }

  handleToggleModal() {
    const { showModal } = this.state;

    this.setState({
      showModal: !showModal,
    });
  }

  getLocationSource() {
    const { location } = this.props;

    return require(`../../assets/map/${lowerCaseRemoveSpaces(location)}.jpg`);
  }

  render() {
    const { showModal } = this.state;
    const { location, dates } = this.props;

    const source = this.getLocationSource();

    return (
      <div>
        <div className="location-header-div">
          <Navbar bg="dark" variant="dark">
            <Navbar.Brand onClick={this.handleToggleModal}>{location}</Navbar.Brand>
            <Navbar.Text>{dates}</Navbar.Text>
            <Navbar.Collapse className="justify-content-end">
              <Nav className="container-fluid mr-auto">
                <LocationNavDropdown />
                <Nav.Link href="/">Home</Nav.Link>
              </Nav>
            </Navbar.Collapse>
          </Navbar>
        </div>
        <MapModal
          handleToggleModal={this.handleToggleModal}
          location={location}
          showModal={showModal}
          source={source}
        />
      </div>
    );
  }
};

LocationHeader.propTypes = {
  location: PropTypes.string.isRequired,
  dates: PropTypes.string.isRequired,
};