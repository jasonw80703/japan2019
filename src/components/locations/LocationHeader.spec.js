import React from 'react';
import renderer from 'react-test-renderer';
import LocationHeader from 'components/locations/LocationHeader';
import { LOCATIONS } from 'constants/locations';
import map from 'assets/map/sanfrancisco.jpg';
import mapModalDetails from 'assets/map/sanfrancisco.json';

describe('<LocationHeader />', () => {

  const defaultProps = {
    location: LOCATIONS.SAN_FRANCISCO,
    map: map,
    mapModalDetails: mapModalDetails,
  };

  it('renders as expected', () => {
    const tree = renderer
      .create(<LocationHeader {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
