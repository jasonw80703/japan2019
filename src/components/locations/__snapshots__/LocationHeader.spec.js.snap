// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationHeader /> renders as expected 1`] = `
<div
  className="location-header-div"
>
  <nav
    className="navbar navbar-expand navbar-light"
    id="navbar-header"
  >
    <span
      className="navbar-brand"
      id="navbar-brand"
      onClick={[Function]}
    >
      San Francisco
    </span>
    <span
      className="navbar-text"
      id="navbar-dates"
      onClick={[Function]}
    >
      Oct 16
    </span>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <div
        className="navbar-nav"
        id="location-nav"
        onKeyDown={[Function]}
      >
        <div
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Location
          </a>
        </div>
        <a
          className="nav-link"
          data-rb-event-key="/sapporo"
          href="/sapporo"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Next
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          id="home-link"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
      </div>
    </div>
  </nav>
</div>
`;
